### PR TITLE
allocator: Vacate now moves AppSearch instances

### LINKED ISSUE #175: AppSearch clusters are ignored in vacate actions
## Expected Behavior

`ecctl platform allocator vacate <allocator ID>` should vacate all the workload in the allocator, including Appsearch.

## Current Behavior

`ecctl platform allocator vacate <allocator ID>` doesn't vacate AppSearch clusters.

### PR DESCRIPTION
**Pull request by marclop**
## Description

Adds support for AppSearch instances to be moved when the `ecctl
platform allocator vacate` is called, adding support for workload
filtering too.

The only difference with other workloads is that plan tracking is not
supported for AppSearch workloads due to elastic/cloud-sdk-go#34.
Instead, an error message is returned after the vacate operations have
been completed which instructrs the user to check the UI to verify that
the vacate has indeed succeeded, of course this is less than ideal, but
once elastic/cloud-sdk-go#34 is resolved, this will no longer be the
case.

## Related Issues

Resolves #175 

## Manual Tests

```console
$ dev-cli platform allocator vacate 192.168.44.11 --kind appsearch
1 error occurred:
	* allocator 192.168.44.11: cluster [716cfd4f68694f5f8c867428c8bf5491][appsearch]: instance is being moved but tracking is not supported, please check the vacate progress from the UI


exit status 255
```

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: pkg/platform/allocator/vacate.go [hunk 1]
```diff
@@ -174,6 +174,8 @@ func waitVacateCompletion(p *pool.Pool, hasWork bool) error {
 	return errs.ErrorOrNil()
 }
 
+// nolint better to have this grouped rather than scattered around, there's
```
Comment (ppapapetrou76) [inline pkg/platform/allocator/vacate.go]: Maybe I haven't got enough ☕️ yet but I don't fully understand this comment here 😊

File: pkg/platform/allocator/vacate.go [hunk 2]
```diff
@@ -384,14 +402,22 @@ func trackMovedCluster(params *VacateClusterParams) error {
 		MaxRetries:    params.MaxPollRetries,
 	})
 	if err != nil {
-		return errors.Wrap(err, vacateWrapMessage(params, "track"))
+		// Remove this after AppSearch is supported in the plan tracker
```
Comment (ppapapetrou76) [inline pkg/platform/allocator/vacate.go]: Maybe this should be a `TODO` pointing to a github issue?

File: pkg/platform/allocator/vacate.go [hunk 3]
```diff
@@ -384,14 +402,22 @@ func trackMovedCluster(params *VacateClusterParams) error {
...
+		return errors.Wrap(err, vacateWrapMessage(params, wrapMessage))
 	}
 
 	return plan.Stream(channel, params.Output)
 }
 
 // CheckVacateFailures iterates over the list of failures returning
 // a multierror with any of the failures found.
+//
+// nolint
```
Comment (ppapapetrou76) [inline pkg/platform/allocator/vacate.go]: What do we want to hide under the carpet here? 😆

File: pkg/platform/allocator/vacate_test.go [hunk 4]
```diff
@@ -599,6 +625,27 @@ func TestCheckVacateFailures(t *testing.T) {
 				errors.New("cluster [123456789][apm] failed vacating, reason: code: unknown, message: an apm error message"),
 			}},
 		},
+		{
+			name: "Returns na apm error on Kibana vacate failure",
```
Comment (ppapapetrou76) [inline pkg/platform/allocator/vacate_test.go]: I believe the test name was born by copy- 🍝 :trollface:

File: pkg/platform/allocator/vacate_test.go [hunk 5]
```diff
@@ -823,6 +870,24 @@ func TestVacateCluster(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Succeeds with a appsearch instance with no tracking",
```
Comment (ppapapetrou76) [inline pkg/platform/allocator/vacate_test.go]:
```suggestion
			name: "Succeeds with an appsearch instance with no tracking",
```

File: pkg/platform/allocator/vacate_test.go [hunk 6]
```diff
@@ -599,6 +625,27 @@ func TestCheckVacateFailures(t *testing.T) {
 				errors.New("cluster [123456789][apm] failed vacating, reason: code: unknown, message: an apm error message"),
 			}},
 		},
+		{
+			name: "Returns na apm error on Kibana vacate failure",
+			args: args{
+				failures: &models.MoveClustersDetails{
+					AppsearchClusters: []*models.MoveAppSearchDetails{
+						{
+							ClusterID: ec.String("123456789"),
```
Comment (ppapapetrou76) [inline pkg/platform/allocator/vacate_test.go]: How this doesn't fail with cluster ID validation. I haven't dived into the test a lot. It just alarmed my brain. Ignore if I'm just drunk

File: pkg/platform/allocator/vacate_test.go (hunk 4 above)
Comment (marclop) [reply pkg/platform/allocator/vacate_test.go]: This test is for CheckVacateFailures, it's not verifying any Cluster IDs

File: pkg/platform/allocator/vacate_test.go (hunk 6 above)
Comment (marclop) [reply pkg/platform/allocator/vacate_test.go]: This test is for CheckVacateFailures, it's not verifying any Cluster IDs

File: pkg/platform/allocator/vacate_test.go (hunk 4 above)
Comment (marclop) [reply pkg/platform/allocator/vacate_test.go]: haha, this made me laugh so much

File: pkg/platform/allocator/vacate.go (hunk 3 above)
Comment (marclop) [reply pkg/platform/allocator/vacate.go]: 👹

Comment (marclop) [reply pkg/platform/allocator/vacate.go]: It's because of the complexity score from the linter

**Pull request merged**